dataland-email-service/src/test/kotlin/org/dataland/datalandemailservice/services/EmailSubscriptionTrackerTest.kt

```diff
@@ -0,0 +1,93 @@
+import org.junit.jupiter.api.Test
+import org.springframework.beans.factory.annotation.Autowired
+import org.springframework.boot.jdbc.EmbeddedDatabaseConnection
+import org.springframework.boot.test.autoconfigure.jdbc.AutoConfigureTestDatabase
+import org.springframework.boot.test.context.SpringBootTest
+import java.util.UUID
+
+@SpringBootTest(classes = [DatalandEmailService::class], properties = ["spring.profiles.active=nodb"])
+@AutoConfigureTestDatabase(connection = EmbeddedDatabaseConnection.H2)
+@Transactional
+class EmailSubscriptionTrackerTest(
+    @Autowired private val emailSubscriptionRepository: EmailSubscriptionRepository,
+) {
+    private lateinit var emailSubscriptionTracker: EmailSubscriptionTracker
+
+    final val subscribedUuid = UUID.randomUUID()
+    final val unsubscribedUuid = UUID.randomUUID()
+
+    final val unknownEmail = "unknown@mail.com"
+    final val subscribedEmail = "subscriber@mail.com"
+    final val unsubscribedEmail = "unsubscriber@mail.com"
+
+    val subscribedEntity =
+        EmailSubscriptionEntity(
+            uuid = subscribedUuid,
+            emailAddress = subscribedEmail,
+            isSubscribed = true,
+        )
+
+    val unsubscribedEntity =
+        EmailSubscriptionEntity(
+            uuid = unsubscribedUuid,
+            emailAddress = unsubscribedEmail,
+            isSubscribed = false,
+        )
+
+    @BeforeEach
+    fun setup() {
+        emailSubscriptionTracker = EmailSubscriptionTracker(emailSubscriptionRepository)
+        emailSubscriptionRepository.deleteAll()
+        emailSubscriptionRepository.save(subscribedEntity)
+        emailSubscriptionRepository.save(unsubscribedEntity)
+    }
+
+    @Test
+    fun `validate if a new subscription entity is created for a unknown email`() {
+        val uuid = emailSubscriptionTracker.insertSubscriptionEntityIfNeededAndReturnUuid(unknownEmail)
+        val newEntity = emailSubscriptionRepository.findByEmailAddress(unknownEmail)
+        if (newEntity != null) {
+            assertTrue(newEntity.isSubscribed, "The email subscription should be unsubscribed.")
+            assertEquals(unknownEmail, newEntity.emailAddress)
+            assertEquals(uuid, newEntity.uuid)
+        }
+    }
+
+    @Test
+    fun `validate if the uuid is returned for a known email`() {
+        val uuid = emailSubscriptionTracker.insertSubscriptionEntityIfNeededAndReturnUuid(subscribedEmail)
+        assertEquals(subscribedUuid, uuid)
+    }
+
+    @Test
+    fun `validate that the uuid  is returned for unsubscriber`() {
+        val uuid = emailSubscriptionTracker.insertSubscriptionEntityIfNeededAndReturnUuid(unsubscribedEmail)
+        assertEquals(unsubscribedUuid, uuid)
+    }
+
+    @Test
+    fun `validate that a subscribed email is returned as true`() {
+        emailSubscriptionTracker.emailIsSubscribed(subscribedEmail)?.let { assertTrue(it) }
+    }
+
+    @Test
+    fun `validate that a unsubscribed email is returned as false`() {
```

Comment:
```suggestion
    fun `validate that an unsubscribed email is returned as false`() {
```

see above; false is returned for an unsubscribed mail